subsys/portability/cmsis_rtos_v2/thread.c

```diff
@@ -509,6 +509,9 @@ __NO_RETURN void osThreadExit(void)
 	__ASSERT(!k_is_in_isr(), "");
 	tid = osThreadGetId();
 
+#if CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT
```

Comment:
`thread_num_dynamic` and `CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT` are always defined, so you can probably write:
```
if (CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT != 0) {
	atomic_dec((atomic_t *)&thread_num_dynamic);
}
```